src/VueGapi/GoogleAuthService.js

```diff
@@ -105,17 +114,50 @@ export default class GoogleAuthService {
     this.authenticated = true
   }
 
+  /**
+   * Returns the authorization code set via {@link GoogleAuthService#grantOfflineAccess}.
+   *
+   * @method GoogleAuthService#getOfflineAccessCode
+   * @return {string|null}
+   */
   getOfflineAccessCode() {
     return this.offlineAccessCode
   }
 
+  /**
+   * Get permission from the user to access the specified scopes offline.
+   *
+   * @method GoogleAuthService#grantOfflineAccess
+   * @see [GoogleAuth.grantOfflineAccess]{@link https://developers.google.com/identity/sign-in/web/reference#googleauthgrantofflineaccessoptions}
+   * @return {Promise}
+   */
   grantOfflineAccess() {
     if (!this.authInstance) throw new Error('gapi not initialized')
     return this.authInstance
       .grantOfflineAccess()
       .then(this._setOfflineAccessCode.bind(this))
   }
 
+  /**
+   * Signs in the user.
+   *
+   * @method GoogleAuthService#login
+   * @see [GoogleAuth.signIn]{@link https://developers.google.com/identity/sign-in/web/reference#googleauthsignin}
+   * @return {Promise}
+   *
+   * @example
+   * <script>
+   *   export default {
+   *     name: 'login-shortcut',
+   *
+   *     methods: {
+   *       login() {
+   *         this.$gapi.login()
+   *       },
+   *     },
+   *   }
+   * </script>
+   */
   login() {
```

Comment:
One of the main issues with the current state of: A lot of these methods are not actually _directly_ exposed on the `Vue` instance, i.e.

https://github.com/cedpoilly/vue-gapi/blob/c276ae549df443bcfc86fe4b02d2327440608af2/src/VueGapi/index.js#L94-L109

actually has a different signature.

@cedpoilly, any thoughts on this? Should we try to move more of the functionality into this `GoogleAuthService` to at least keep the method signatures the same?

Reply:
Still open to suggestions, but I took a pass at this in 1927c3d596596383c5eb99fedc2cf596d6e7f01b.